tests/unit/multi-currency/test-class-currency-switcher-widget.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Class WCPay_Multi_Currency_Currency_Switcher_Widget_Tests
+ *
+ * @package WooCommerce\Payments\Tests
+ */
+
+use WCPay\Multi_Currency\Currency;
+
+/**
+ * WCPay\Multi_Currency\Currency_Switcher_Widget unit tests.
+ */
+class WCPay_Multi_Currency_Currency_Switcher_Widget_Tests extends WP_UnitTestCase {
+	/**
+	 * Mock WCPay\Multi_Currency\Multi_Currency.
+	 *
+	 * @var WCPay\Multi_Currency\Multi_Currency|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_multi_currency;
+
+	/**
+	 * WCPay\Multi_Currency\Currency_Switcher_Widget instance.
+	 *
+	 * @var WCPay\Multi_Currency\Currency_Switcher_Widget
+	 */
+	private $currency_switcher_widget;
+
+	/**
+	 * Pre-test setup
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->mock_multi_currency = $this->createMock( WCPay\Multi_Currency\Multi_Currency::class );
+		$this->mock_multi_currency
+			->method( 'get_enabled_currencies' )
+			->willReturn(
+				[
+					new Currency( 'USD' ),
+					new Currency( 'CAD', 1.2 ),
+					new Currency( 'EUR', 0.8 ),
+				]
+			);
+
+		$this->currency_switcher_widget = new WCPay\Multi_Currency\Currency_Switcher_Widget( $this->mock_multi_currency );
+	}
+
+	public function test_widget_output() {
+		$args = [
+			'before_title'  => '<h2>',
+			'after_title'   => "</h2>\n",
+			'before_widget' => '<section>',
+			'after_widget'  => "</section>\n",
+		];
+
+		$instance = [
+			'title' => 'Test Title',
+		];
+
+		ob_start();
```

Comment:
PHP Unit provides some methods of testing the output of the program, which are used in [some of our tests](https://github.com/Automattic/woocommerce-payments/blob/4876fc5b2384388691ecd4b95e7835b4b70009a6/tests/unit/test-class-wc-payment-gateway-wcpay.php#L316-L335). They can be a bit limited in regards to testing multiple things, but what do you think of using a similar approach in these tests?

Reply:
Oh! Didn't know about that. I used some code from WP to see how output was tested. I'll dig a bit in how we are testing to update the code. Thanks for pointing that out! 🙂

Reply:
Both the rewrite to use `expectOutputRegex` and split of widget addressed in 400f4c2